feature/bgp/policybase/otg_tests/nested_policies/nested_policies_test.go

```diff
@@ -234,146 +235,220 @@ func TestBGPNestedPolicies(t *testing.T) {
 	if err != nil {
-		t.Fatalf("AppendNewStatement(%s) failed: %v", v4LPStatement, err)
-	}
-	if !deviations.SkipSettingStatementForPolicy(dut) {
-		t.Logf("Setting statement for policy")
-		stmt1.GetOrCreateActions().SetPolicyResult(oc.RoutingPolicy_PolicyResultType_ACCEPT_ROUTE)
+		t.Fatalf("AppendNewStatement(%s) failed: %v", "20", err)
 	}
-	stmt1.GetOrCreateActions().GetOrCreateBgpActions().SetSetLocalPref(localPref)
+	stmt.GetOrCreateActions().PolicyResult = oc.RoutingPolicy_PolicyResultType_ACCEPT_ROUTE
 
-	// Configure a route-policy to match the prefix.
-	pdef2 := rp.GetOrCreatePolicyDefinition(v4PrefixPolicy)
-	stmt2, err := pdef2.AppendNewStatement(v4PrefixStatement)
-	if err != nil {
-		t.Fatalf("AppendNewStatement(%s) failed: %v", v4PrefixStatement, err)
-	}
-	if !deviations.SkipSettingStatementForPolicy(dut) {
-		stmt2.GetOrCreateActions().SetPolicyResult(oc.RoutingPolicy_PolicyResultType_ACCEPT_ROUTE)
-	}
+	if operation == "set" {
+		// Configure a route-policy to set the local preference.
+		pdef1 := rp.GetOrCreatePolicyDefinition(v4LPPolicy)
+		stmt1, err := pdef1.AppendNewStatement(v4LPStatement)
+		if err != nil {
+			t.Fatalf("AppendNewStatement(%s) failed: %v", v4LPStatement, err)
+		}
+		if !deviations.SkipSettingStatementForPolicy(dut) {
+			t.Logf("Setting statement for policy")
+			stmt1.GetOrCreateActions().SetPolicyResult(oc.RoutingPolicy_PolicyResultType_ACCEPT_ROUTE)
+		}
+		stmt1.GetOrCreateActions().GetOrCreateBgpActions().SetSetLocalPref(localPref)
+
+		// Configure a route-policy to match the prefix.
+		pdef2 := rp.GetOrCreatePolicyDefinition(v4PrefixPolicy)
+		stmt2, err := pdef2.AppendNewStatement(v4PrefixStatement)
+		if err != nil {
+			t.Fatalf("AppendNewStatement(%s) failed: %v", v4PrefixStatement, err)
+		}
+		if !deviations.SkipSettingStatementForPolicy(dut) {
+			stmt2.GetOrCreateActions().SetPolicyResult(oc.RoutingPolicy_PolicyResultType_ACCEPT_ROUTE)
+		}
 
-	t.Logf("Configuring nested policy")
-	// Configure a prefix-set for route filtering/matching.
-	prefixSet := rp.GetOrCreateDefinedSets().GetOrCreatePrefixSet(v4PrefixSet)
-	prefixSet.SetMode(oc.PrefixSet_Mode_IPV4)
-	prefixSet.GetOrCreatePrefix(advertisedIPv41.cidr(t), maskLenExact)
+		t.Logf("Configuring nested policy")
+		// Configure a prefix-set for route filtering/matching.
+		prefixSet := rp.GetOrCreateDefinedSets().GetOrCreatePrefixSet(v4PrefixSet)
+		prefixSet.SetMode(oc.PrefixSet_Mode_IPV4)
+		prefixSet.GetOrCreatePrefix(advertisedIPv41.cidr(t), maskLenExact)
 
-	if !deviations.SkipSetRpMatchSetOptions(dut) {
-		stmt2.GetOrCreateConditions().GetOrCreateMatchPrefixSet().SetMatchSetOptions(oc.RoutingPolicy_MatchSetOptionsRestrictedType_ANY)
+		if !deviations.SkipSetRpMatchSetOptions(dut) {
+			stmt2.GetOrCreateConditions().GetOrCreateMatchPrefixSet().SetMatchSetOptions(oc.RoutingPolicy_MatchSetOptionsRestrictedType_ANY)
+		}
+		stmt2.GetOrCreateConditions().GetOrCreateMatchPrefixSet().SetPrefixSet(v4PrefixSet)
+	}
+	if operation == "set" {
```

Comment:
Why do we need multiple if else for operation "set" and "delete".
Can we club them up with the if operation == "set" at line 251
	
	if operation == "set" {
		gnmi.BatchReplace(batch, gnmi.OC().RoutingPolicy().Config(), rp)
	} else if operation == "delete" {
		gnmi.BatchDelete(batch, gnmi.OC().RoutingPolicy().Config())
		gnmi.BatchReplace(batch, gnmi.OC().RoutingPolicy().Config(), rp)
	}
	dni := deviations.DefaultNetworkInstance(dut)
	if operation == "set" {
		// Configure the nested policy.
		rpPolicy := root.GetOrCreateRoutingPolicy()
		statPath := rpPolicy.GetOrCreatePolicyDefinition(v4LPPolicy).GetStatement(v4LPStatement).GetOrCreateConditions()
		statPath.SetCallPolicy(v4PrefixPolicy)
		gnmi.BatchReplace(batch, gnmi.OC().RoutingPolicy().Config(), rpPolicy)
	} else if operation == "delete" {
		gnmi.BatchDelete(batch, gnmi.OC().RoutingPolicy().Config())
		gnmi.BatchReplace(batch, gnmi.OC().RoutingPolicy().Config(), rp)
	}

Reply:
Updated the code to remove multiple if else condition.